File: apps/api/wr3_api/domain/enums.py
from enum import StrEnum


class Chain(StrEnum):
    ETHEREUM = "ethereum"
    BASE = "base"
    BSC = "bsc"
    ARBITRUM = "arbitrum"
    SOLANA = "solana"


class AuditState(StrEnum):
    CREATED = "created"
    QUEUED = "queued"
    INGESTING = "ingesting"
    NEEDS_SOURCE = "needs_source"
    STATIC_RUNNING = "static_running"
    TRIAGE_RUNNING = "triage_running"
    POC_RUNNING = "poc_running"
    FUZZING_RUNNING = "fuzzing_running"
    SCORING = "scoring"
    HUMAN_REVIEW = "human_review"
    CHANGES_REQUESTED = "changes_requested"
    PARTIAL = "partial"
    COMPLETED = "completed"
    FAILED = "failed"
    RETRYING = "retrying"
    REJECTED = "rejected"
    TERMINAL = "terminal"


class Severity(StrEnum):
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"
    INFO = "info"


class Exploitability(StrEnum):
    CONFIRMED = "confirmed"
    LIKELY = "likely"
    THEORETICAL = "theoretical"
    UNKNOWN = "unknown"
    DISMISSED = "dismissed"


class PocStatus(StrEnum):
    NOT_ATTEMPTED = "not_attempted"
    FAILED = "failed"
    CONFIRMED = "confirmed"


class HumanReviewStatus(StrEnum):
    NOT_REQUIRED = "not_required"
    PENDING = "pending"
    APPROVED = "approved"
    REJECTED = "rejected"


class Tier(StrEnum):
    FREE = "free"
    HOBBY = "hobby"
    TEAM = "team"
    PRO = "pro"


class RequestedDepth(StrEnum):
    PRELIMINARY = "preliminary"
    STANDARD = "standard"
    DEEP = "deep"


class Visibility(StrEnum):
    PRIVATE = "private"
    PUBLIC = "public"


class UserIntent(StrEnum):
    PRE_LAUNCH_SELF_CHECK = "pre_launch_self_check"
    THIRD_PARTY_RESEARCH = "third_party_research"
    MONITORING = "monitoring"

File: packages/types/src/enums.ts
export const chains = ["ethereum", "base", "bsc", "arbitrum", "solana"] as const;
export const auditStates = [
  "created",
  "queued",
  "ingesting",
  "needs_source",
  "static_running",
  "triage_running",
  "poc_running",
  "fuzzing_running",
  "scoring",
  "human_review",
  "changes_requested",
  "partial",
  "completed",
  "failed",
  "retrying",
  "rejected",
  "terminal"
] as const;
export const severities = ["critical", "high", "medium", "low", "info"] as const;
export const exploitabilities = ["confirmed", "likely", "theoretical", "unknown", "dismissed"] as const;
export const tiers = ["free", "hobby", "team", "pro"] as const;
export const requestedDepths = ["preliminary", "standard", "deep"] as const;
export const visibilities = ["private", "public"] as const;
export const userIntents = ["pre_launch_self_check", "third_party_research", "monitoring"] as const;
export const humanReviewStatuses = ["not_required", "pending", "approved", "rejected"] as const;
export const pocStatuses = ["not_attempted", "failed", "confirmed"] as const;

export type Chain = (typeof chains)[number];
export type AuditState = (typeof auditStates)[number];
export type Severity = (typeof severities)[number];
export type Exploitability = (typeof exploitabilities)[number];
export type Tier = (typeof tiers)[number];
export type RequestedDepth = (typeof requestedDepths)[number];
export type Visibility = (typeof visibilities)[number];
export type UserIntent = (typeof userIntents)[number];
export type HumanReviewStatus = (typeof humanReviewStatuses)[number];
export type PocStatus = (typeof pocStatuses)[number];
